cluster-autoscaler/main.go

```diff
@@ -177,6 +178,13 @@ var (
 	clusterAPICloudConfigAuthoritative = flag.Bool("clusterapi-cloud-config-authoritative", false, "Treat the cloud-config flag authoritatively (do not fallback to using kubeconfig flag). ClusterAPI only")
 	cordonNodeBeforeTerminate          = flag.Bool("cordon-node-before-terminating", false, "Should CA cordon nodes before terminating during downscale process")
 	daemonSetEvictionForEmptyNodes     = flag.Bool("daemonset-eviction-for-empty-nodes", false, "DaemonSet pods will be gracefully terminated from empty nodes")
+
+	initialNodeGroupBackoffDuration = flag.Duration("initial-node-group-backoff-duration", clusterstate.InitialNodeGroupBackoffDuration,
```

Comment:
Let's remove constants from clusterstate and put them here.

Reply:
seems  we can not , some unit tests also dependent on the constants.

Reply:
I see that the parameters are used to initialize backoff in tests in two packages: core and  clusterstate.

In principle the default parameters shouldn't be the only ones we're testing. The tests should be written in a way that allows using any valid set of parameters. Frequently in these situations dummy values like "1 minute" are used.

We are where we are with the current tests, but I see no reason they should be using the default values specifically. WDYT about changing the parameters to literals/local non-exported constants in these 2 packages, and moving the defaults to the flags themselves? We could change the test parameters to some dummy values if feasible, but leaving the current values also SGTM if the tests are depending on the specific values.

Compare this with e.g. `scaleDownCandidatesPoolMinCount` - the default value `50` is in the flag, and different tests are using different values for it. Or `unremovableNodeRecheckTimeout` - the default value `5*time.Minute` is in the flag, and tests are just using the `5*time.Minute` literal.